salt/fileclient.py

```diff
@@ -462,6 +462,18 @@ def get_url(self, url, dest, makedirs=False, saltenv='base',
         '''
         url_data = urlparse(url)
 
+        # If dest is a directory, rewrite dest with filename
+        if os.path.isdir(dest) or dest.endswith(('/', '\\')):
+            if url_data.query or len(url_data.path) > 1 and not url_data.path.endswith('/'):
+                strpath = url.split('/')[-1]
+            else:
+                strpath = 'index.html'
```

Comment:
My original request was to not have to specifiy a destination name. cp supports pulling from http, so where should we save in the case of `salt windowsmin cp.get_url 'http://www.google.com/' C:\\` ? In wget, it uses 'index.html' if there is no query string or file path. Well, actually Im sure wget supports a lot more cases (301, etc)..

To me, its not super practical (who wants to save an index.html page??) but its a case all the same..